source/server/ScriptEngine.cpp

```diff
@@ -68,1156 +68,1405 @@ _CRTIMP void __cdecl _wassert(_In_z_ const wchar_t * _Message, _In_z_ const wcha
-    assert_net(result >= 0);
-    result = engine->RegisterEnumValue("authType", "RoRnet::AUTH_ALL", 0xFFFFFFFF);
-    assert_net(result >= 0);
-
-    // Register serverSayType
-    result = engine->RegisterEnum("serverSayType");
-    assert_net(result >= 0);
-    result = engine->RegisterEnumValue("serverSayType", "FROM_SERVER", FROM_SERVER);
-    assert_net(result >= 0);
-    result = engine->RegisterEnumValue("serverSayType", "FROM_HOST", FROM_HOST);
-    assert_net(result >= 0);
-    result = engine->RegisterEnumValue("serverSayType", "FROM_MOTD", FROM_MOTD);
-    assert_net(result >= 0);
-    result = engine->RegisterEnumValue("serverSayType", "FROM_RULES", FROM_RULES);
-    assert_net(result >= 0);
-
-    // register constants
-    result = engine->RegisterGlobalProperty("const int TO_ALL", (void *) &TO_ALL);
-    assert_net(result >= 0);
-
-
-    Logger::Log(LOG_INFO, "ScriptEngine: Registration done");
-}
-
-void ScriptEngine::msgCallback(const asSMessageInfo *msg) {
-    const char *type = "Error";
-    if (msg->type == asMSGTYPE_INFORMATION)
-        type = "Info";
-    else if (msg->type == asMSGTYPE_WARNING)
-        type = "Warning";
-
-    Logger::Log(LOG_INFO, "ScriptEngine: %s (%d, %d): %s = %s", msg->section, msg->row, msg->col, type, msg->message);
+void ScriptEngine::init()
+{
+   int result;
+
+   // Create the script engine
+   engine = asCreateScriptEngine(ANGELSCRIPT_VERSION);
+
+   // Set the message callback to receive information on errors in human readable form.
+   // It's recommended to do this right after the creation of the engine, because if
+   // some registration fails the engine may send valuable information to the message
+   // stream.
+   result = engine->SetMessageCallback(asMETHOD(ScriptEngine, msgCallback), this, asCALL_THISCALL);
+   if (result < 0)
+   {
+      if (result == asINVALID_ARG)
+      {
+         Logger::Log(LOG_ERROR,
+                     "ScriptEngine: One of the arguments is incorrect, e.g. obj is null for a class method.");
+         return;
+      }
+      else if (result == asNOT_SUPPORTED)
+      {
+         Logger::Log(LOG_ERROR, "ScriptEngine:  The arguments are not supported, e.g. asCALL_GENERIC.");
+         return;
+      }
+      Logger::Log(LOG_ERROR, "ScriptEngine: Unknown error while setting up message callback");
+      return;
+   }
+
+   // AngelScript doesn't have a built-in string type, as there is no definite standard
+   // string type for C++ applications. Every developer is free to register it's own string type.
+   // The SDK do however provide a standard add-on for registering a string type, so it's not
+   // necessary to register your own string type if you don't want to.
+   RegisterStdString(engine);
+   RegisterScriptArray(engine,
+                       true);      // true = allow arrays to be registered as type[] (e.g. int[]). Needed for backwards compatibillity.

```

Comment:
This call was probably line-broken because of the long comment. 
If you wouldn't mind the comment spanning far to the right (I wouldn't), just remove the linebreak please.
Otherwise, rewrite like this:
```
const bool allow_cstyle_array_syntax = true; // e.g. int[]; Needed for backwards compatibility
RegisterScriptArray(engine, allow_cstyle_array_syntax);
```